datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/MonthEnumDeserializerModifier.java

```diff
@@ -0,0 +1,60 @@
+package com.fasterxml.jackson.datatype.jsr310.deser;
+
+import java.io.IOException;
+import java.time.Month;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.util.JacksonFeatureSet;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DeserializationConfig;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerModifier;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeFeature;
+
+
+public class MonthEnumDeserializerModifier extends BeanDeserializerModifier {
+    private final boolean _oneBaseMonths;
+
+    public MonthEnumDeserializerModifier() {
+        this(false);
+    }
+
+    public MonthEnumDeserializerModifier(boolean oneBaseMonths) {
+        _oneBaseMonths = oneBaseMonths;
+    }
+
+    public MonthEnumDeserializerModifier withFeatures(JacksonFeatureSet<JavaTimeFeature> features) {
+        boolean enabled = features.isEnabled(JavaTimeFeature.ONE_BASED_MONTHS);
+        if (enabled == _oneBaseMonths) {
+            return this;
+        }
+        return new MonthEnumDeserializerModifier(enabled);
+    }
+
+    @Override
+    public JsonDeserializer<?> modifyEnumDeserializer(DeserializationConfig config, JavaType type, BeanDescription beanDesc, JsonDeserializer<?> deserializer) {
+        if(type.getRawClass() != Month.class) {
+            return deserializer;
+        }
+        return new JsonDeserializer<Enum>() {
```

Comment:
Ok, this is too simplistic: my idea was to extend `DelegatingDeserializer` passing the default deserializer and overriding its `deserialize()` method -- ONLY handling special case of 1-based month and number, otherwise calling `super.deserialize()` (`getDelegatee().deserialize(...)` .

Reply:
I have applied your suggestions and a few other small changes. Can you take a look when you'll have some time?
Also, I haven't tackled the part about serialization yet - will we follow the same approach?